pkg/authz/authz.go

```diff
@@ -23,3 +82,139 @@ func NewAuthorizer(config *Config, server ServerInterface, logger logger.Logger)
 		logger: logger,
 	}
 }
+
+func (a *Authorizer) getRelation(apiMethod string) (string, error) {
+	switch apiMethod {
+	case ReadAuthorizationModel, ReadAuthorizationModels:
+		return CanCallReadAuthorizationModels, nil
+	case Read:
+		return CanCallRead, nil
+	case Write:
+		return CanCallWrite, nil
+	case ListObjects, StreamedListObjects:
+		return CanCallListObjects, nil
+	case Check:
+		return CanCallCheck, nil
+	case ListUsers:
+		return CanCallListUsers, nil
+	case WriteAssertions:
+		return CanCallWriteAssertions, nil
+	case ReadAssertions:
+		return CanCallReadAssertions, nil
+	case WriteAuthorizationModel:
+		return CanCallWriteAuthorizationModels, nil
+	case ListStores:
+		return CanCallListStores, nil
+	case CreateStore:
+		return CanCallCreateStore, nil
+	case GetStore:
+		return CanCallGetStore, nil
+	case DeleteStore:
+		return CanCallDeleteStore, nil
+	case Expand:
+		return CanCallExpand, nil
+	case ReadChanges:
+		return CanCallReadChanges, nil
+	default:
+		return "", fmt.Errorf("unknown api method: %s", apiMethod)
+	}
+}
+
+// AuthorizeCreateStore checks if the user has access to create a store.
+func (a *Authorizer) AuthorizeCreateStore(ctx context.Context, clientID string) (bool, error) {
+	relation, err := a.getRelation(CreateStore)
+	if err != nil {
+		return false, err
+	}
+	allowed, err := a.individualAuthorize(ctx, clientID, relation, a.getSystem(), &openfgav1.ContextualTupleKeys{})
+	if !allowed || err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+// ListAuthorizedStores returns the list of stores that the user has access to.
+func (a *Authorizer) ListAuthorizedStores(ctx context.Context, clientID string) ([]string, error) {
```

Comment:
This is odd to me, the method above is merely checking is the clientID can call CreateStore, but here you are checking if clientID can call ListStores _and_ also executing that call, this last part seems out of scope for an authorizer

Reply:
Will fix this function to just return `(bool, error)` and move the list stores logic out.